Stack/Opc.Ua.Core/Stack/Server/EndpointBase.cs

```diff
@@ -848,6 +939,19 @@ private void OnProcessRequest(object state)
                     // set the context.
                     SecureChannelContext.Current = m_context;
 
+                    // check if AdditionalHeader field from the request contains trace information
+                    if (m_request.RequestHeader.AdditionalHeader != null && m_request.RequestHeader.AdditionalHeader.Body is AdditionalParametersType parameters)
+                    {
+                        // extract trace information from the request header
+                        var traceContext = ExtractTraceContextFromParameters(parameters);
+
+                        // create activity using the method name as the display name
+                        var activity = ActivitySource.StartActivity(m_request.GetType().Name, ActivityKind.Server, traceContext.ActivityContext);
+
```

Comment:
looks strange to me that in the Start we provide already the parent and now it has to be set again?